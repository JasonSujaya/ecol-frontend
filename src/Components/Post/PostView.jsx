import React from "react";
import Grid from "@material-ui/core/Grid";
import Box from "@material-ui/core/Box";
import PostShowBookmark from "./PostTitle/PostShowBookmark";
import { get_post_feed } from "./postview.js";
import PostTitle from "./PostTitle/PostTitleView";
import Createpost from "./CreatePost/CreatePostView";

class PostView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      postList: [],
    };
  }
  putPostInState = async () => {
    console.log("updated");
    let data = await get_post_feed().catch((err) =>
      console.log("Response body", err.response.data)
    );

    this.setState({ postList: data });
  };

  async componentDidMount() {
    this.putPostInState();
  }

  renderPost = () => {
    return this.state.postList.length == 0 ? (
      <div className="loading">"Loading"</div>
    ) : (
      this.state.postList.reverse().map((item, index) => {
        return (
          <PostTitle
            id={this.state.postList[index].id}
            data={this.state.postList[index]}
            key={this.state.postList[index].id}
          ></PostTitle>
        );
      })
    );
  };

  allowPost = () => {
    return this.props.loggedInTrue ? (
      <Createpost putPostInState={this.putPostInState}></Createpost>
    ) : (
      <div>HIII</div>
    );
  };

  render() {
    return (
      <div>
        {this.allowPost()}
        <Box mt="2rem">
          <Grid container spacing={1} direction="column">
            <Grid item container>
              <Grid item xs={1} sm={1} md={1}></Grid>
              <Grid item xs={12} sm={9} md={7} className="hi">
                <div className="rendered"> {this.renderPost()}</div>
              </Grid>
              <Grid item xs={1} sm={1} md={3}>
                <PostShowBookmark></PostShowBookmark>
              </Grid>
            </Grid>
          </Grid>
        </Box>
      </div>
    );
  }
}

export default PostView;
