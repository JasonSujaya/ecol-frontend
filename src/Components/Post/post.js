import axios from "axios";

export const get_post_feed = async () => {
  const response = await axios.get(
    `https://ecole-heroku.herokuapp.com/api/post-api/post-manager/`
  );
  const data = response.data;
  return data;
};

export const get_category = async () => {
  const response = await axios.get(
    `https://ecole-heroku.herokuapp.com/api/post-api/post-category/`
  );
  return response;
};

export const create = async (_title, _content, _category) => {
  const user = {
    title: _title,
    content: _content,
    category: _category,
    tag: [
      {
        tag_id: 1,
      },
    ],
    images: [],
  };

  const response = await axios
    .post(
      `https://ecole-heroku.herokuapp.com/api/post-api/post-manager/`,
      user,
      {
        headers: {
          authorization: `Token ${localStorage.token}`,
          "Content-Type": "application/json",
          Accept: "application/json",
        },
      }
    )
    .then()
    .catch((err) => console.log("Response body", err.response.data));
  console.log("Triggered");
  console.log(response);

  return response;
};

export const getCategoryName = (inputId) => {
  if (inputId == 1) {
    return "Category1";
  } else if (inputId == 2) {
    return "Category2";
  }
};
