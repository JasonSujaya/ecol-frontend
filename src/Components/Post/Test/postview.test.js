import axios from "axios";
import { shallow, mount } from "enzyme";
import { get_post_feed } from "../post";

import PostView from "../PostView.js";

const runAllPromises = () => new Promise(setImmediate);
function tick() {
  return new Promise((resolve) => {
    setTimeout(resolve, 0);
  });
}

const setUp = () => {
  axios.get.mockResolvedValue({
    data: [
      {
        bookmarks: 0,
        category: 1,
        content: "Here is a post",
        id: 1,
        images: [],
        post_bookmarkslist: [],
        post_comments: [],
        tag: [],
        title: "My First Album",
        user: 1,
      },
      {
        bookmarks: 0,
        category: 1,
        content: "This is another post",
        id: 2,
        images: [],
        post_bookmarkslist: [],
        post_comments: [],
        tag: [],
        title: "My Latest Album",
        user: 1,
      },
    ],
  });
};

describe("Post View Component", () => {
  it("PostView renders correctly when promise is not resolved", () => {
    // Arrange
    setUp();

    //Act
    const wrapper = mount(<PostView />);

    //Assert
    expect(wrapper).toMatchSnapshot();
  });

  it("PostView renders correctly when promise is resolved", async () => {
    // Arrange
    setUp();

    //Act
    const wrapper = mount(<PostView />);
    await runAllPromises();
    wrapper.update();

    //Assert
    expect(wrapper).toMatchSnapshot();
  });
  it("Returns the title of the first post should be correct", async () => {
    // Arrange
    setUp();

    // Act
    const title = await get_post_feed();

    // Assert
    expect(title[0].title).toEqual("My First Album");
  });

  it("Renders loading when promise is not resolved", () => {
    // Arrange
    setUp();

    //Act
    const wrapper = mount(<PostView />);

    //Assert
    expect(wrapper.find(".loading")).toHaveLength(1);
  });

  it("Renders posttitleview component correctly when promise is resolved", async () => {
    // Arrange
    setUp();
    const wrapper = await mount(<PostView />);

    //Act
    await runAllPromises();
    wrapper.update();

    //Assert
    expect(wrapper.find(".postTitleBox")).toHaveLength(2);
    expect(wrapper.find(".postTitle").first().text()).toContain(
      "My Latest Album"
    );
    expect(wrapper.find(".postTitle").at(2).text()).toContain("My First Album");
  });
});
