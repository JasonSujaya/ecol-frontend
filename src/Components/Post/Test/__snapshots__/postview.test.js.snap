// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post View Component PostView renders correctly when promise is not resolved 1`] = `ReactWrapper {}`;

exports[`Post View Component PostView renders correctly when promise is resolved 1`] = `ReactWrapper {}`;
